src/main/java/Decide.java

```diff
@@ -79,6 +79,22 @@ public boolean LIC3() throws IllegalArgumentException {
         return false;
     }
 
+    /**
+     * Check if LIC4 is true.
+     *
+     * @return true if there exists at least one set of Q_PTS consecutive data points
+     * that lie in more than QUADS quadrants, false otherwise.
+     */
+    public boolean LIC4() throws IllegalArgumentException {
+        if (parameters.getQ_PTS() < 2) {
+            throw new IllegalArgumentException("Q_PTS must be greater than or equal to 2.");
```

Comment:
New try, I forgot to comment in the code. I am still learning :). Is IllegalArgumentException a good choice? The function takes no arguments.

Reply:
No, since the method doesn't take any arguments. Created a custom exception called IllegalParameterObjectException and changed all the places where IllegalArgumentException was used to use this new exception.